dbt/adapters/athena/column.py

```diff
@@ -59,4 +80,18 @@ def data_type(self) -> str:
         if self.is_timestamp():
             return self.timestamp_type()
 
+        if self.is_array():
+            # Resolve the inner type of the array, using an AthenaColumn
+            # instance to properly convert the inner type. Note that this will
+            # cause recursion in cases of nested arrays
+            inner_type = self.array_inner_type()
+            inner_type_col = AthenaColumn(
+                column=self.column,
+                dtype=inner_type,
+                char_size=self.char_size,
+                numeric_precision=self.numeric_precision,
+                numeric_scale=self.numeric_scale,
+            )
```

Comment:
Instantiating an `AthenaColumn` instance on the inner type feels a little bit hacky, but it's downstream of the fact that the type parsing logic is tightly coupled to the `AthenaColumn` class definition. We could eliminate the need to instantiate another `AthenaColumn` object here by decoupling the parsing logic from the class, but that felt like a bigger refactor to me, so I'll only go down that road if you think it's worth it.